components/board/genericlinux/hw_pwm.go

```diff
@@ -0,0 +1,173 @@
+// Package genericlinux is for Linux boards. This particular file is for using sysfs to
+// interact with PWM devices. All of these functions are idempotent: you can double-export a pin or
+// double-close it with no problems.
+package genericlinux
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type pwmDevice struct {
+	// These values are immutable
+	chipPath string
+	line     int
+
+	mu sync.Mutex
+
+	// These values are mutable
+	periodNs         uint64
+	activeDurationNs uint64
+	isExported       bool
+	isEnabled        bool
+}
+
+func NewPwmDevice(chipName string, line int) (*pwmDevice, error) {
+	// There should be a single directory within /sys/devices/platform/<chipName>/pwm/, whose name
+	// is mirrored in /sys/class/pwm. That's the one we want to use.
```

Comment:
Yes, this sounds pretty janky. but it's the same approach NVidia took in their own Python library: https://github.com/NVIDIA/jetson-gpio/blob/master/lib/python/Jetson/GPIO/gpio_pin_data.py#L577-L598